accounts/views.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.views import APIView
+from accounts.serializers import *
+from django.contrib.auth import authenticate
+from accounts.utils import Util
+from django.core.mail import send_mail
+
+class UserRegistrationView(APIView):
+    def post(self, request, format=None):
+        serializer = UserRegistrationSerializer(data=request.data)
+        if serializer.is_valid():   
+            user = serializer.save() # otp_created_at is created when this is saved
+            user.otp = Util.generateOTP()
+            user.save()
+            # Send an email with the verification code
+            subject = 'Activate your account'
+            message = f'Your verification code is: {user.otp}'
+            from_email = 'admin@gmail.com'
+            recipient_list = [user.email]
+
+            send_mail(subject, message, from_email, recipient_list, fail_silently=False)
+            return Response({'msg':'OTP for verification has been sent'}, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    
+class VerifyEmailView(APIView):
+     def post(self, request, format=None):
+        serializer = VerifyEmailSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
rise exception if serializer is not valid

Reply:
The validate function in the VerifyEmailSeriallizer does the exception handling.
`def validate(self, attrs):
        email = attrs.get('email')
        OTP = attrs.get('otp')
        #Checking if email exists or not
        if User.objects.filter(email=email).exists():
            user = User.objects.get(email=email)
            otp_expiry_time = user.otp_created_at + timedelta(minutes=5)  
            #otp expires if it preceeds 5 min
            if timezone.now() > otp_expiry_time:
                raise serializers.ValidationError('OTP has expired')
            #verify otp
            if not user.otp == OTP :
                raise serializers.ValidationError('OTP does not match')
        else:
            raise serializers.ValidationError('Email does not exist')
        user.is_verified = True
        user.otp = None
        user.save()
        return attrs`

Reply:
may be it will rise exception ?